ocrd_segment/repair.py

```diff
@@ -399,3 +345,18 @@ def _plausibilize_group(regionspolys, rogroup, mark_for_deletion, mark_for_mergi
         if region.parent_object_:
             # remove in-place
             region.parent_object_.get_TextRegion().remove(region)
+
+def make_valid(polygon):
```

Comment:
ocrd_utils.image?

Reply:
You mean to cover all of the above in OCR-D/core#492 or OCR-D/core#577? I am not so sure – remember we actually wanted to fix this at the level of the PAGE model (whenever adding elements), throw a specialized exception in each unique case, collect page-wise exceptions in Processor.process_pages, turning them into error messages, and opt-in to seeing the stack traces with `--debug`.

Reply:
> remember we actually wanted to fix this at the level of the PAGE model

I am aware of and working on that as time permits. But since you have written these pretty generic utility methods like `make_valid(polygon)`, I still think it would be useful to offer them to all the OCR-D projects. Just an offer, proceed as you think is best.

Reply:
A function like `make_valid` is too low-level IMHO. So far we don't pass/return Shapely objects anywhere. As for `polygon_for_parent`, I don't think it would be a good investment of resources to rewrite other processors with that stopgap, too. We should instead concentrate on the encapsulated, automatic solution in core – as time permits of course.